File: 1544-MakeTheStringGreat/1544-MakeTheStringGreat.py
class Solution:
    def makeGood(self, s: str) -> str:
        #create a stack
        stack = []
        
        # loop through the string
        for c in s:
            # check if the current string converted to uppercase or lowercase is equal to the last item in the stack
            if stack and c.swapcase() == stack[-1]:
                #remove the last item of the stack
                stack.pop()
            else:
                #add it to the stack
                stack.append(c)
        
        #join the stack
        return ''.join(stack)

File: 1544-MakeTheStringGreat/1544-MakeTheStringGreat.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
    let stack = [];

    function swapCase(s){
         return s === s.toLowerCase() ? s.toUpperCase() : s.toLowerCase();
    }

    s.split('').forEach(c => {
        if(stack.length > 0 && stack[stack.length - 1] == swapCase(c)){
            stack.pop();
        } else {
            stack.push(c);
        }
    })
    
    return stack.join('');
};
